.github/workflows/build-and-test.yaml

```diff
@@ -65,12 +65,46 @@ jobs:
           -DCMAKE_CXX_COMPILER=clang++
         cmake --build . --target install -- -j$(nproc)
 
+  prepare-e2e-test:
+    name: Prepare e2e test
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout EmitC
+      uses: actions/checkout@v2
+      with:
+        path: ${{ env.EMITC }}
+        submodules: 'false'
+
+    - name: Cache e2e
+      id: cache-e2e
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.E2E }}
+        key: ${{ runner.os }}-e2e-${{ hashFiles('emitc/scripts/*.py') }}
+
+    - name: Install TensorFlow
+      if: steps.cache-e2e.outputs.cache-hit != 'true'
+      run: python3 -m pip install tensorflow==2.4
```

Comment:
We can move the dependencies to a requirements.txt file. This makes it easier to reproduce for users.

Reply:
Yeah, I intend to add some documentation on our e2e flow anyway, but this is a good point.